src/csharp/Grpc.Auth/GoogleAuthInterceptors.cs

```diff
@@ -43,29 +45,22 @@ public static class GoogleAuthInterceptors
         /// <returns>The interceptor.</returns>
         public static AsyncAuthInterceptor FromCredential(ITokenAccess credential)
         {
-            if(credential is ITokenAccessWithExtraInformation credentialWithExtra)
-            {
-                return FromCredential(credentialWithExtra);
-            }
-            
-            return new AsyncAuthInterceptor(async (context, metadata) =>
+            AsyncAuthInterceptor interceptors = new AsyncAuthInterceptor(async (context, metadata) => 
             {
-                await AddBearerTokenHeader(credential, context, metadata).ConfigureAwait(false);
+                var accessToken = await credential.GetAccessTokenForRequestAsync(context.ServiceUrl, CancellationToken.None).ConfigureAwait(false);
+                metadata.Add(CreateBearerTokenHeader(accessToken));
             });
-        }
 
-        /// <summary>
-        /// Creates an <see cref="AsyncAuthInterceptor"/> that will obtain access token and associated information
-        /// from any credential type that implements <c>ITokenAccessWithExtraInformation</c>.
-        /// </summary>
-        /// <param name="credential">The credential to use to obtain access tokens.</param>
-        /// <returns>The interceptor.</returns>
-        public static AsyncAuthInterceptor FromCredential(ITokenAccessWithExtraInformation credential)
-        {
-            return new AsyncAuthInterceptor(async (context, metadata) => {
-                await AddBearerTokenHeader(credential, context, metadata).ConfigureAwait(false);
-                AddExtraInforHeaders(credential, context, metadata);
-            });
+            if (credential is IProjectImpersonatorTokenAccess impersonator)
+            {
+                interceptors += new AsyncAuthInterceptor((context, metadata) => 
+                {
+                    metadata.Add(CreateHeader(QuotaProjectIdHeader, impersonator.QuotaProjectId));
```

Comment:
My general concern about `IProjectImpersonatorTokenAccess` is that it's limited to a single functionality and header name. Problems with that:
- if more headers like this are added (could always happen), we will need to make the same hack again
- like this we need to duplicate the Google auth logic in Grpc.Auth (remember the name of the header, a specific interface etc.)

Reply:
Yep, this I don't like either.
On the other hand with the ITokenAccessWithExtraInformation implementation we are risking information not related to auth piggy backing on the token.
I don't know if you saw [the ITokenAccessWithExtraInformation](https://github.com/amanda-tarafa/grpc/pull/1/commits/fb5a2992740703beeea972369c26f181f8b906ff) commit here, given that I had the dirty commits on the PR (I already fixed that). If you haven't, take a look at that proposal and let me know if you prefer it.

Reply:
Yes, I saw the other proposal (ITokenAccessWithExtraInformation) and I like it better (as described in the other PR).